src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -232,9 +231,16 @@ class VLMPipeline::VLMPipelineImpl : public VLMPipelineBase{
 
         auto start_get_inputs_embeds = std::chrono::steady_clock::now();
         if (m_inputs_embedder->has_token_type_ids()) {
-            std::tie(inputs_embeds, token_type_ids) = m_inputs_embedder->get_inputs_embeds_with_token_type_ids(unified_prompt, encoded_images, perf_metrics, encoded_images.size() > 0, image_sequence);
+            std::tie(inputs_embeds, token_type_ids) =
+                m_inputs_embedder->get_inputs_embeds_with_token_type_ids(unified_prompt,
+                                                                         encoded_images,
+                                                                         encoded_videos,
+                                                                         perf_metrics,
+                                                                         encoded_images.size() > 0 || encoded_videos.size() > 0,
+                                                                         image_sequence,
+                                                                         video_sequence);
         } else {
-            inputs_embeds = m_inputs_embedder->get_inputs_embeds(unified_prompt, encoded_images, encoded_videos, perf_metrics, encoded_images.size() > 0, image_sequence, video_sequence);
+            inputs_embeds = m_inputs_embedder->get_inputs_embeds(unified_prompt, encoded_images, encoded_videos, perf_metrics, encoded_images.size() > 0 || encoded_videos.size() > 0, image_sequence, video_sequence);
```

Comment:
Let's move `encoded_images.size() > 0 || encoded_videos.size() > 0` to separate bool variable (e.g. `recalculate_merged_embeddings`) and reuse

Reply:
Done.